### PR TITLE
Fix weather effects not applying to areas like space

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request
Weather effects were ignoring certain areas like space. This was due to checking the area's `z` position. Some areas like `area/space` are in many different z-levels and the `z` position defaults to Centcomm `z=1`. 

The solution is to loop through `area.turfs_by_zlevel` and check if it has any turfs in that z-level.

CC @LemonInTheDark the mutable overlay for weather effects is not applying to space.

https://github.com/tgstation/tgstation/blob/7e27663517731fe8f3d955477b1a97ace5a6ff83/code/datums/weather/weather.dm#L253-L271

I'm assuming this is due to plane master shengians. I could just add the effects like the radiation nebula:

https://github.com/tgstation/tgstation/blob/7e27663517731fe8f3d955477b1a97ace5a6ff83/code/datums/station_traits/negative_traits.dm#L486-L493

But this doesn't appear reversible. I'm open to suggestions.

## Why It's Good For The Game
Rad storms now affect space outside the station. Other weather effects should also be consistent and not be ignored by certain misc. areas types.

## Changelog
:cl:
fix: Fix weather effects ignoring certain areas like space. 
/:cl:

### CODE REVIEW

File: code/datums/weather/weather.dm [hunk 1]
```diff
@@ -102,16 +102,19 @@
...
+		affectareas += selected_area
+	for(var/area/protected_area as anything in protected_areas)
+		affectareas -= get_areas(protected_area)
+	for(var/area/affected_area as anything in affectareas)
+		if(protect_indoors && !affected_area.outdoors)
 			continue
-		if(A.z in impacted_z_levels)
-			impacted_areas |= A
+
+		for(var/z as anything in impacted_z_levels)
```
Comment (MrMelbert) [inline code/datums/weather/weather.dm:113]:
```suggestion
		for(var/z in impacted_z_levels)
```

**Comment by LemonInTheDark** [comment]
I'm not sure we actually want this. space is more of a grouping area rather then a concrete "thing" and weather effects can't really be z limited, radstorms on all 12 z levels seems weird.

**Comment by timothymtorres** [comment]
So we are trying to avoid people going into space to avoid radiation storms like so:

<details>
<summary>Solar Array During Rad Storm</summary>

dreamseeker_5AylD0BtTI

</details>

I just tested the weather effects on space ruins and it was not affected by a radiation storm set to the station z-level. I'm guessing there is a check somewhere in the weather code to only apply the effect if they are on the same z-level it's being triggered on.

<details>
<summary>Space Ruin During Rad Storm</summary>

dreamseeker_nYFBWSi6kQ

</details>

**Comment by LemonInTheDark** [comment]
the behavior could be z limited but the visuals never could be, and damage without visuals would not be fair either.

**Comment by timothymtorres** [comment]
So I found the cause:

https://github.com/tgstation/tgstation/blob/0b0114cb39ccd59264ab9103ab69e1de5cdb2a29/code/game/area/areas.dm#L178-L196

`static_lighting` -> `blend_mode` -> `BLEND_MULTIPLY` 

When you disable `BLEND_MULTIPLY` we can use overlays on area/space. Now to fix this, perhaps I should set the `blend_mode` to `area/space` to default values and then switch back to `BLEND_MULTIPLY` after the weather effect.

**Comment by LemonInTheDark** [comment]
@Ghommie why did you merge this I was discussing how it would break visuals in new and interesting ways

**Comment by timothymtorres** [comment]
> I'm not sure we actually want this. space is more of a grouping area rather then a concrete "thing" and weather effects can't really be z limited, radstorms on all 12 z levels seems weird. can't make up mym ind

The solution I found addressed this. The radstorms only hurt and appear on the station z level for space areas. I didn't have to mess with the parallax layer.

> @Ghommie why did you merge this I was discussing how it would break visuals in new and interesting ways

I did mention in my prior reply that I fixed the problem so he probably felt that was directed at your review.

**Comment by LemonInTheDark** [comment]
whoops edited your post instead of replying real win of a day

**Comment by timothymtorres** [comment]
> How? 

The weather act has a check to make sure it only happens on the selected z-levels:

https://github.com/tgstation/tgstation/blob/07ee08cf5e180ca678b966831a5d05dd1b830ce8/code/datums/weather/weather.dm#L195-L206

> also ftr if modifying the blendmode does nothing it maybe doesn't need one, I have a feeling it only exists because of an old, outdated implementation of area lighting

I wouldn't be surprised.

**Comment by LemonInTheDark** [comment]
> The weather act has a check to make sure it only happens on the selected z-levels:

how does this help with visuals

**Pull request merged**